File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int n = nums.size(), currentSum = 0, minSubArr = INT_MAX;
        int i = 0;
        for(int j = 0; j < n; ++j) {
            currentSum += nums[j];
            while(currentSum >= target) {
                minSubArr = min(minSubArr, j - i + 1);
                currentSum -= nums[i];
                i++;
            }
        }
        return minSubArr == INT_MAX ? 0 : minSubArr; 
    }
};

File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n, currentSum, minSubArr, i = len(nums), 0, 1e7, 0
        for j in range(n):
            currentSum += nums[j];
            while currentSum >= target: 
                minSubArr = min(minSubArr, j - i + 1);
                currentSum -= nums[i];
                i += 1
        if minSubArr == 1e7: return 0
        else: return minSubArr